src/algorithms/util/matrixToListConverter.java

```diff
@@ -0,0 +1,35 @@
+package src.algorithms.util;
+import java.util.*;
+/**
+ * Supports method to convert an Adjacency Matrix into an Adjacency list
+ * Adjacency Matrix in the form of a 2D Integer Array int[][]
+ * Adjacency list in the form of a 2D Integer array int[][], where array[i] is a list of neighbour indexes
+ */
+
+public class matrixToListConverter {
+
+    public int[][] convert(int[][] adjM) {
+
+        List<ArrayList<Integer>> adjL = new ArrayList<>();
+        int numOfNodes = adjM.length;
+
+        for (int i = 0; i < numOfNodes; i++) {
+            adjL.add(new ArrayList<>());
+            for (int j = 0; j < numOfNodes; j++) {
+                if (adjM[i][j] == 1) {
+                    adjL.get(i).add(j);
+                }
+            }
+        }
+
+        int[][] res = new int[numOfNodes][];
+        for (int i = 0; i < numOfNodes; i++) {
+            ArrayList<Integer> neighbourList = adjL.get(i);
+            for (int j = 0; j < neighbourList.size(); j++) {
```

Comment:
yup, there's prolly no need to re-add here as well, based on what was mentioned in the comments above. Makes implementation succinct too.